### PR TITLE
[DDW-1109] Allow specifying Node.js version; patch `electron-rebuild` forcing proper Electron-Node.js headers; fix M1 CI

### PR DESCRIPTION
**Pull request by michalrus**
Following numerous complains around `electron-rebuild` and mismatched Node.js ABI version of native modules (@danielmain, @renanvalentin, @tomislavhoracek, …), I am:

* downgrading Node.js to 14.17.0 (allowing specifying the version easily, just [edit this line](https://github.com/input-output-hk/daedalus/blob/54e576cf10233b4ae7dfdc3ae87896100359d7b9/default.nix#L74)),
* patching `electron-rebuild` in Yarn’s `postInstall` — to force correct Electron–Node.js headers to build against,
* adding `nodejs` to Hydra, to have it cached for others — it takes a significant amount of time to build,
* running all CI tasks with the same Nix 2.5,
* turning off Nix garbage collection in M1 (`aarch64-darwin`) CI, which seems to fix flaky segfaults we were experiencing (also @DominikGuzei reports that `export GC_DONT_GC=1` fixes his local segfault issue).

## Todos

- [x] Linux `yarn dev`
- [x] Linux installer
- [x] macOS `yarn dev`
- [x] macOS installer
- [x] Windows installer

## Screenshots

n/a

## Testing Checklist

- [Slack QA thread](https://input-output-rnd.slack.com/archives/GGKFXSKC6/p1655144042729409)
- [ ] Test

---

### Basics

- [x] PR assigned to the PR author(s)
- [x] `input-output-hk/daedalus-dev` and `input-output-hk/daedalus-qa` assigned as PR reviewers
- [x] If there are UI changes, Alexander Rukin assigned as an additional reviewer
- [ ] All visual regression testing has been reviewed (assign `run Chromatic` label to PR to trigger the run)
- [x] PR has appropriate labels (`release-vNext`, `feature`/`bug`/`chore`, `WIP`)
- [x] PR link is added to a Jira ticket, ticket moved to In Review
- [x] PR is updated to the most recent version of the target branch (and there are no conflicts)
- [x] PR has a good description that summarizes all changes
- [ ] PR contains screenshots (in case of UI changes)
- [ ] CHANGELOG entry has been added to the top of the appropriate section (_Features_, _Fixes_, _Chores_) and is linked to the correct PR on GitHub
- [ ] There are no missing translations (running `yarn manage:translations` produces no changes)
- [ ] Text changes are proofread and approved (Jane Wild / Amy Reeve)
- [ ] Japanese text changes are proofread and approved (Junko Oda)
- [ ] Storybook works and no stories are broken (`yarn storybook`)
- [ ] In case of dependency changes `yarn.lock` file is updated

### Code Quality

- [ ] Important parts of the code are properly commented and documented
- [ ] Code is properly typed with typescript types
- [ ] React components are split-up enough to avoid unnecessary re-renderings
- [ ] Any code that only works in main process is neatly separated from components

### Testing

- [ ] New feature/change is covered by acceptance tests
- [ ] New feature/change is manually tested and approved by QA team
- [ ] All existing acceptance tests are still up-to-date
- [ ] New feature/change is covered by Daedalus Testing scenario
- [ ] All existing Daedalus Testing scenarios are still up-to-date

### After Review

- [ ] Update Slack QA thread by marking it with a green checkmark

### CODE REVIEW

File: release.nix [hunk 1]
```diff
@@ -69,8 +69,11 @@ in {
   in {
   daedalus-installer = allArchesNoWindows;
   yaml2json = allArchesNoWindows;
+  nodejs = allArchesNoWindows;
```
Comment (michalrus) [inline release.nix:72]: Let’s cache this in Hydra, since building from source takes some considerable time.

**Review by tomislavhoracek — APPROVED** [review]
💯

**Review by danielmain — APPROVED** [review]
Great Job @michalrus

**Comment by michalrus** [comment]
@danielmain, awesome, thank you! :blush:

**Pull request merged**